components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketHandler.java

```diff
@@ -0,0 +1,437 @@
+					infoDTO.setValidationStatus(APIConstants.KeyValidationStatus.API_BLOCKED);
+					infoDTO.setType(type);
+					infoDTO.setAuthorized(false);
+					return false;
+				}
+
+				String apiProvider = rs.getString("API_PROVIDER");
+				String subTier = rs.getString("TIER_ID");
+				String appTier = rs.getString("APPLICATION_TIER");
+				infoDTO.setTier(subTier);
+				infoDTO.setSubscriber(rs.getString("USER_ID"));
+				infoDTO.setApplicationId(rs.getString("APPLICATION_ID"));
+				infoDTO.setApiName(rs.getString("API_NAME"));
+				infoDTO.setApiPublisher(apiProvider);
+				infoDTO.setApplicationName(rs.getString("NAME"));
+				infoDTO.setApplicationTier(appTier);
+				infoDTO.setType(type);
+
+				//Advanced Level Throttling Related Properties
+				if (APIUtil.isAdvanceThrottlingEnabled()) {
+					String apiTier = rs.getString("API_TIER");
+					String subscriberUserId = rs.getString("USER_ID");
+					String subscriberTenant = MultitenantUtils.getTenantDomain(subscriberUserId);
+					int apiId = rs.getInt("API_ID");
+					int subscriberTenantId = APIUtil.getTenantId(subscriberUserId);
+					int apiTenantId = APIUtil.getTenantId(apiProvider);
+					boolean isContentAware =
+							isAnyPolicyContentAware(conn, apiTier, appTier, subTier,
+							                        subscriberTenantId, apiTenantId, apiId);
+					infoDTO.setContentAware(isContentAware);
+
+					int spikeArrest = 0;
+					String apiLevelThrottlingKey = "api_level_throttling_key";
+					if (rs.getInt("RATE_LIMIT_COUNT") > 0) {
+						spikeArrest = rs.getInt("RATE_LIMIT_COUNT");
+					}
+
+					String spikeArrestUnit = null;
+					if (rs.getString("RATE_LIMIT_TIME_UNIT") != null) {
+						spikeArrestUnit = rs.getString("RATE_LIMIT_TIME_UNIT");
+					}
+					boolean stopOnQuotaReach = rs.getBoolean("STOP_ON_QUOTA_REACH");
+					List<String> list = new ArrayList<String>();
+					list.add(apiLevelThrottlingKey);
+					infoDTO.setSpikeArrestLimit(spikeArrest);
+					infoDTO.setSpikeArrestUnit(spikeArrestUnit);
+					infoDTO.setStopOnQuotaReach(stopOnQuotaReach);
+					infoDTO.setSubscriberTenantDomain(subscriberTenant);
+					if (apiTier != null && apiTier.trim().length() > 0) {
+						infoDTO.setApiTier(apiTier);
+					}
+					//We also need to set throttling data list associated with given API. This need to have policy id and
+					// condition id list for all throttling tiers associated with this API.
+					infoDTO.setThrottlingDataList(list);
+				}
+				return true;
+			}
+			infoDTO.setAuthorized(false);
+			infoDTO.setValidationStatus(
+					APIConstants.KeyValidationStatus.API_AUTH_RESOURCE_FORBIDDEN);
+		} catch (SQLException e) {
+			handleException("Exception occurred while validating Subscription.", e);
+		} finally {
+			try {
+				conn.setAutoCommit(false);
+			} catch (SQLException e) {
+
+			}
+			APIMgtDBUtil.closeAllConnections(ps, conn, rs);
+		}
+		return false;
+	}
+
+	private boolean isAnyPolicyContentAware(Connection conn, String apiPolicy, String appPolicy,
```

Comment:
These methods are already there in DAO classes. Any reason to add these methods new?